lib_xud/src/core/included/XUD_Token_In_DI.S

```diff
@@ -133,27 +154,29 @@ TxHandShakeReceived:
 #else
     ldc        r9, USB_PIDn_ACK
 #endif
-    xor        r9, r11, r9
-    bt         r9, BadHandshake
+    {clre;              sub r9, r11, r9} 
```

Comment:
Delaying disabling events "feels" naughty. Maybe you can reason it's safe though and no other events may fire during this period.

Reply:
// No other events are enabled here, both RxV and TxRdIn are cleared before, so can delay clre to this point